### ISSUE TITLE
R Version Date Missing 

### ISSUE THREAD

**Issue by jasmyace** [tmcd82070/CAMP_RST#55]
(from Doug)

One small issue I noticed is that your 10 04 2015 version of R is missing something.  I say that because if you use the Platform interface, and click on the “Versions” button in the primary dashboard with the data entry, pre analysis, and data analysis buttons, the version of R is no longer reported (see screen shot below).  Please refresh your memory on where the date of the R code is stored in the R files so the next version of the R code can report that date.  Having that date helps isolate where problems may be occurring.

![image003](https://cloud.githubusercontent.com/assets/9057972/11961387/6fe26604-a895-11e5-8718-4f1a8b2c7187.jpg)

**Comment by jasmyace** [comment]
I will look to see where this data piece is needed in the code, and make the change for the next release.

**Comment by dougthreloff** [comment]
Jason:



Thanks for looking into the issue of how to make sure the R code has
something that indicates what version is being used when a person is in the
Platform user interface.



Doug



*From:* Jason [mailto:notifications@github.com]
*Sent:* Tuesday, December 22, 2015 9:19 AM
*To:* tmcd82070/CAMP_RST
*Subject:* [CAMP_RST] R Version Date Missing (#55)



(from Doug)

One small issue I noticed is that your 10 04 2015 version of R is missing
something. I say that because if you use the Platform interface, and click
on the “Versions” button in the primary dashboard with the data entry, pre
analysis, and data analysis buttons, the version of R is no longer reported
(see screen shot below). Please refresh your memory on where the date of
the R code is stored in the R files so the next version of the R code can
report that date. Having that date helps isolate where problems may be
occurring.

[image: Image removed by sender. image003]
<https://cloud.githubusercontent.com/assets/9057972/11961387/6fe26604-a895-11e5-8718-4f1a8b2c7187.jpg>

—
Reply to this email directly or view it on GitHub
<https://github.com/tmcd82070/CAMP_RST/issues/55>.[image: Image removed by
sender.]

**Comment by jasmyace** [comment]
I think maybe we're using different Platform versions.  When I hit the "Version" button on mine, I get a pop-up that makes no mention of R files:

![image](https://cloud.githubusercontent.com/assets/9057972/11965476/021065c8-a8b2-11e5-9652-915ff103dec8.png)

Or, maybe there's some other reason as to why our displays are different?  Has there been an update to the Platform? 

For the record, I've never set an R version before in the code, so if this feature has been updating itself over time since I've been here, I did nothing to initiate those changes.  I wonder if this is a recent development, or if maybe this functionality was turned off when we updated to the new version of R internally to the Platform? 

If this is something that was present, but whose functionality has since disappeared, the updating of R would be my guess as to the culprit.  Anything else you might know / recall historically about this would be useful in remedying this issue.

**Comment by dougthreloff** [comment]
Jason:



The two screenshots below illustrate what I am seeing.  The one providing 3
versions and 3 dates illustrates what I hope to see.  The other screenshot
shows the missing “R analysis files version” date when I load version 4.4.
of the R code that was meant to deal with a problem with the Mokelumne
River data.  I am guessing Trent was sticking something in his code, or he
had a file with a date that did not change each time the platform was
opened, that helped a user know which version of the R files people had
loaded for use.



Doug











*From:* Jason [mailto:notifications@github.com]
*Sent:* Tuesday, December 22, 2015 12:49 PM
*To:* tmcd82070/CAMP_RST
*Cc:* dougthreloff
*Subject:* Re: [CAMP_RST] R Version Date Missing (#55)



I think maybe we're using different Platform versions. When I hit the
"Version" button on mine, I get a pop-up that makes no mention of R files:

[image: Image removed by sender. image]
<https://cloud.githubusercontent.com/assets/9057972/11965476/021065c8-a8b2-11e5-9652-915ff103dec8.png>

Or, maybe there's some other reason as to why our displays are different?
Has there been an update to the Platform?

For the record, I've never set an R version before in the code, so if this
feature has been updating itself over time since I've been here, I did
nothing to initiate those changes. I wonder if this is a recent
development, or if maybe this functionality was turned off when we updated
to the new version of R internally to the Platform?

If this is something that was present, but whose functionality has since
disappeared, the updating of R would be my guess as to the culprit.
Anything else you might know / recall historically about this would be
useful in remedying this issue.

—
Reply to this email directly or view it on GitHub
<https://github.com/tmcd82070/CAMP_RST/issues/55#issuecomment-166726154>.[image:
Image removed by sender.]

**Comment by MikeBanach** [comment]
All,
    
    When you click the "Version" button on the main form the following
    information is reported by the user interface.  This information is
    gathered and reported by the user interface program and does not
    come from R or from the R-Interface files.
    
    User interface version:
    This comes from the version of the interface application.  It is
    hard-coded in the interface software.
    
    Database version:
    This is fetched from the Version field of the Version table in the
    database.  (That table should have only a single record.)
    
    R-Interface version:
    If there is no zip file in the R-Interface directory then the
    following should be reported:
         R analysis files version:  << not
      found >>
    
    If there is a single zip file in the R-Interface directory then that
    file name is reported.
         R analysis files version:  [file
        name reported] 
    
    If there more than one zip files are in the R-Interface directory
    then the following should be reported.
         R analysis files version:  <<
      multiple versions found >>
    
    R version:
    The version of R is not reported.  (This is the R software
    itself, not the analysis routines written by Trent/Jason.)
    
    
    Jason, because your output does not include the R-Interface version,
    maybe you are using an older version of the user interface?  Doug,
    you may want to test to be sure the output you get matches the rules
    I mentioned above for the R-Interface version.  If something doesn't
    work correctly then I can make any fix that may be needed.
    
    Mike
    
    Michael Banach
StreamNet Fisheries Biologist
Pacific States Marine Fisheries Commission
503-595-3152         mike_banach@psmfc.org
www.streamnet.org

    
      
      
      On 12/22/2015 1:20 PM, dougthreloff wrote:
    
    
      
      Jason:
      
      
      
      The two screenshots below illustrate what I am seeing. The one
      providing 3
      versions and 3 dates illustrates what I hope to see. The other
      screenshot
      shows the missing “R analysis files version” date when I load
      version 4.4.
      of the R code that was meant to deal with a problem with the
      Mokelumne
      River data. I am guessing Trent was sticking something in his
      code, or he
      had a file with a date that did not change each time the platform
      was
      opened, that helped a user know which version of the R files
      people had
      loaded for use.
      
      
      
      Doug
      
      
      
      
      
      
      
      
      
      
      
      *From:* Jason [mailto:notifications@github.com]
      *Sent:* Tuesday, December 22, 2015 12:49 PM
      *To:* tmcd82070/CAMP_RST
      *Cc:* dougthreloff
      *Subject:* Re: [CAMP_RST] R Version Date Missing (#55)
      
      
      
      I think maybe we're using different Platform versions. When I hit
      the
      "Version" button on mine, I get a pop-up that makes no mention of
      R files:
      
      [image: Image removed by sender. image]
<https://cloud.githubusercontent.com/assets/9057972/11965476/021065c8-a8b2-11e5-9652-915ff103dec8.png>
      
      Or, maybe there's some other reason as to why our displays are
      different?
      Has there been an update to the Platform?
      
      For the record, I've never set an R version before in the code, so
      if this
      feature has been updating itself over time since I've been here, I
      did
      nothing to initiate those changes. I wonder if this is a recent
      development, or if maybe this functionality was turned off when we
      updated
      to the new version of R internally to the Platform?
      
      If this is something that was present, but whose functionality has
      since
      disappeared, the updating of R would be my guess as to the
      culprit.
      Anything else you might know / recall historically about this
      would be
      useful in remedying this issue.
      
      —
      Reply to this email directly or view it on GitHub
<https://github.com/tmcd82070/CAMP_RST/issues/55#issuecomment-166726154>.[image:
      Image removed by sender.]
      —
        Reply to this email directly or view
          it on GitHub.

**Comment by jasmyace** [comment]
Thanks Mike, this is helpful.  So...

1.  The version of the Platform I'm using is the one I can readily find on the `ftp` site, or  `CAMPRSTcode20140923.zip`.

    **Doug** or **Connie**:  Is there a newer version of the Platform I should be using?  I'm certain Trent is using the same version I am.  
![image](https://cloud.githubusercontent.com/assets/9057972/12156494/67dc26d0-b488-11e5-9252-eb9162aece2c.png)
2.  It sounds like taking any file, zipping it, and then renaming the zip file to whatever needs to be reported in the Platform is the solution, after making sure to place it in the `R-Interface` folder. 

    **Doug**:  does this work?  My version of the Platform doesn't readily report the `R-Interface` version, so I can't check.  But I may be able to, if we can resolve No. 1 above.

**Comment by dougthreloff** [comment]
Mike:  thanks for shedding light on what it takes to make the version of
the R code visible in the Platform user interface’s “version” button.



Jason:  Here are the versions I am using (and I hope you are using the
same):  user interface:  2015.7.6.1300    and back end database structure:
20130610.  I am currently using the following version of R code:  2015 06
17 but will upgrade that when you give me the new R code that addresses the
task 2 changes.



In regards to your bullet #2 below, if I understand what Mike said, the
version tab is picking up on the date in the folder name of the zipped
folder; if that is the case, the zipped folder name you create with the new
R files should have a syntax with the date when you last changed any of the
R code files.  To test things, I took a zipped R interface folder you gave
me that did not have a date in the folder name, renamed the folder with a
date, then opened the platform’s version tab and the version tab did show
the version of the R interface folder with my new date, so Mike’s approach
works.



I think I have noticed that some of the R file dates in the unzipped folder
change as a user opens and closes the platform, so the dates on those R
files cannot be used to determine which version of the R code is being used
because they change each time someone uses the platform.



Doug









*From:* Jason [mailto:notifications@github.com]
*Sent:* Wednesday, January 06, 2016 2:24 PM
*To:* tmcd82070/CAMP_RST
*Cc:* dougthreloff
*Subject:* Re: [CAMP_RST] R Version Date Missing (#55)



Thanks Mike, this is helpful. So...

1.     The version of the Platform I'm using is the one I can readily find
on the ftp site, or CAMPRSTcode20140923.zip.

*Doug* or *Connie*: Is there a newer version of the Platform I should be
using? I'm certain Trent is using the same version I am.

[image: Image removed by sender. image]
<https://cloud.githubusercontent.com/assets/9057972/12156494/67dc26d0-b488-11e5-9252-eb9162aece2c.png>

2.     It sounds like taking any file, zipping it, and then renaming the
zip file to whatever needs to be reported in the Platform is the solution,
after making sure to place it in the R-Interface folder.

*Doug*: does this work? My version of the Platform doesn't readily report
the R-Interface version, so I can't check. But I may be able to, if we can
resolve No. 1 above.

—
Reply to this email directly or view it on GitHub
<https://github.com/tmcd82070/CAMP_RST/issues/55#issuecomment-169481855>.[image:
Image removed by sender.]
